packages/webpack-cli/lib/utils/warnings/bailAndWatchWarning.js

```diff
@@ -0,0 +1,7 @@
+const logger = require('../logger');
+
+module.exports = (compiler) => {
+    if (compiler.options.bail && compiler.watch) {
```

Comment:
Looks good, but why we don't use `compiler.watchMode`?

Reply:
Don't know why but it isn't working with `compiler.watchMode`.

Reply:
https://github.com/webpack/webpack/blob/master/lib/Compiler.js#L336 is always `true`, we need to use `watchMode`, can you try it again?

Reply:
Not working. I logged `compiler` in the console and found `watchMode: false` instead 😕 , 

![screenshot-github-com-webpack-webpack-cli-pull-1804-1600434195860](https://user-images.githubusercontent.com/46647141/93600842-e7301c80-f9dd-11ea-892d-2116b606d366.png)

Reply:
oh, yes, I was wrong, `watchMode` is true only when we run `compiler.watch()`, so we need use `watch`

Reply:
Maybe we should use `compiler.options.watch`?

Reply:
with `compiler.options.watch` the process exits too soon without printing the warning in case an error is found.

![Screenshot at 2020-09-18 18-54-11](https://user-images.githubusercontent.com/46647141/93602826-cc12dc00-f9e0-11ea-98f8-9c0253f012ae.png)

Reply:
We need to search way to detect watch mode, because `compiler.watch` is method and it is always true

Reply:
Ah yes, then we have `compiler.options.bail && (outputOptions.watch || options.watch)`

Reply:
`outputOptions.watch` is `true` when we run `compiler.watch`

Reply:
> `compiler.options.bail && (outputOptions.watch || compiler.options.watch)` 

`outputOptions,watch` is true when we pass `--watch` via `CLI`, so we need to check the existence of `outputOptions.watch` else `webpack --bail --watch` will not log the warning. The rest of all cases will log the warning.

Reply:
Fixed, the issue was with `--watch` flag it was not setting `compiler.options.watch` equal to `true`. Now works perfect.

![Screenshot at 2020-09-19 15-37-23](https://user-images.githubusercontent.com/46647141/93664638-14d49e80-fa8e-11ea-9653-ea0d25e18017.png)